File: core/src/ru/myitschool/dcrawler/effects/FloorClearingEffect.java
package ru.myitschool.dcrawler.effects;

import com.badlogic.gdx.graphics.Texture;

/**
 * Created by Voyager on 29.11.2017.
 */
public class FloorClearingEffect extends Effect {

    private String name = "";
    private String description = "";
    private Texture icon = null;
    private boolean positive = true;
    FloorEffect effect;
    private int turns;

    public FloorClearingEffect(FloorEffect effect, int turns) {
        super(null);
        setHide(true);
        this.effect = effect;
        this.turns = turns;
    }

    @Override
    public Texture getIcon() {
        return icon;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public boolean isSkillUse() {
        return false;
    }

    @Override
    public boolean isPositive() {
        return true;
    }

    @Override
    public boolean isStackable() {
        return true;
    }

    @Override
    public int getStackSize() {
        return 4096;
    }

    @Override
    public boolean isExpiring() {
        return true;
    }

    @Override
    public int getExpireTurns() {
        return turns;
    }

    @Override
    public String getType() {
        return "main.dcrawler.effect.floor_clearing";
    }

    @Override
    public void onExpire() {
        super.onExpire();
        effect.removeEffect(true);
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/FloorClearingEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture

/**
 * Created by Voyager on 29.11.2017.
 */
class FloorClearingEffect(effect: FloorEffect, turns: Int) : Effect(null) {
    override val name = ""
    override val description = ""
    override val icon: Texture = Texture("bomb.png")
    override val positive: Boolean = true
    override val skillUse: Boolean = false
    override val stackable: Boolean = true
    override val stackSize: Int = 4096
    override val expiring: Boolean = true
    override val expireTurns: Int = turns
    override val id: String = "main.dcrawler.effect.floor_clearing"
    private var effect: FloorEffect

    override fun onExpire() {
        super.onExpire()
        effect.removeEffect(true)
    }

    init {
        hide = true
        this.effect = effect
    }
}